src/hb-font.cc

```diff
@@ -1854,10 +1870,25 @@ hb_font_set_variations (hb_font_t *font,
   if (unlikely (coords_length && !normalized))
     return;
 
+  float *design_coords = coords_length ? (float *) calloc (coords_length, sizeof (float)) : nullptr;
+  if (unlikely (coords_length && !design_coords))
+  {
+    free (normalized);
+    return;
+  }
```

Comment:
Same here.  Combine the allocation-failure checking.

Reply:
Done